src/Mod/Fem/femsolver/elmer/equations/electrostatic_writer.py

```diff
@@ -125,22 +125,26 @@ def handleElectrostaticBndConditions(self):
                     # output the FreeCAD label as comment
                     if obj.Label:
                         self.write.boundary(name, "! FreeCAD Name", obj.Label)
-                    if obj.PotentialEnabled:
-                        if hasattr(obj, "Potential"):
-                            # Potential was once a float and scaled not fitting SI units
-                            if isinstance(obj.Potential, float):
-                                savePotential = obj.Potential
-                                obj.removeProperty("Potential")
-                                obj.addProperty(
-                                    "App::PropertyElectricPotential",
-                                    "Potential",
-                                    "Parameter",
-                                    "Electric Potential",
-                                )
-                                # scale to match SI units
-                                obj.Potential = savePotential * 1e6
-                            potential = float(obj.Potential.getValueAs("V"))
-                            self.write.boundary(name, "Potential", potential)
+                    if obj.Dirichlet:
+                        if obj.PotentialEnabled:
+                            if hasattr(obj, "Potential"):
+                                # Potential was once a float and scaled not fitting SI units
+                                if isinstance(obj.Potential, float):
+                                    savePotential = obj.Potential
+                                    obj.removeProperty("Potential")
+                                    obj.addProperty(
+                                        "App::PropertyElectricPotential",
+                                        "Potential",
+                                        "Parameter",
+                                        "Electric Potential",
+                                    )
+                                    # scale to match SI units
+                                    obj.Potential = savePotential * 1e6
+                                potential = float(obj.Potential.getValueAs("V"))
+                                self.write.boundary(name, "Potential", potential)
+                    elif not obj.Dirichlet and hasattr(obj, "SurfaceChargeDensity"):
```

Comment:
This line looks a bit strange to me as this else is already depending on `obj.Dirichlet`, hence you only need to check check the second part. Ie. You can change this line to:
```python
elif hasattr(obj, "SurfaceChargeDensity"):
```

If you want to keep `not obj.Dirichlet`, then you can change the first if chains at lines 128-130 to:
```python
if obj.Dirichlet and obj.PotentialEnabled and hasattr(obj, "Potential"):
```

and this one to just:
```python
if not obj.Dirichlet and hasattr(obj, "SurfaceChargeDensity"):
```